File: app/src/main/java/com/expensivebelly/dagger2retaingraph/fragment/ComponentComposite.java
package com.expensivebelly.dagger2retaingraph.fragment;

import com.expensivebelly.dagger2retaingraph.fragment.detail.di.DetailComponent;
import com.expensivebelly.dagger2retaingraph.fragment.master.di.MasterComponent;

public interface ComponentComposite {

    MasterComponent masterComponent();

    DetailComponent detailComponent();
}

File: app/src/main/java/com/expensivebelly/dagger2retaingraph/fragment/ComponentComposite.kt
package com.expensivebelly.dagger2retaingraph.fragment

import com.expensivebelly.dagger2retaingraph.fragment.detail.di.DetailComponent
import com.expensivebelly.dagger2retaingraph.fragment.master.di.MasterComponent

interface ComponentComposite {
    val masterComponent: MasterComponent
    val detailComponent: DetailComponent
}